bundles/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/internal/handler/BigAssFanHandler.java

```diff
@@ -1081,17 +1090,20 @@ protected synchronized void disconnect() {
             logger.debug("Disconnecting from {} at {}", thing.getUID(), ipAddress);
 
             try {
-                if (fanWriter != null) {
-                    fanWriter.close();
+                DataOutputStream localFanWriter = fanWriter;
+                if (localFanWriter != null) {
+                    localFanWriter.close();
```

Comment:
Should it still be assigned after this? Otherwise we could risk close() being called again.

Reply:
From the docs it is not very clear what happens when you call close to an allready closed :
_When not already closed, the close method of FilterOutputStream calls its flush method, and then calls the close method of its underlying output stream._

Similar for the `Socket.Close()` method 9 lines lower. That is also not clear. To be sure i assigned null